### PR TITLE
Implement new epic article count opt out journey

### PR DESCRIPTION
**Pull request by Aracho1**
## What does this change?
- In line with the new epic article count opt out feature on support-dotcom-rendering, this PR links CMP to the contributions epic and checks whether a reader has opted out of article count

### CODE REVIEW

File: install_nvm.sh [hunk 1]
```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bash
```
Comment (tomrf1) [inline install_nvm.sh:1]: this file can be deleted!

File: scripts/nginx/Brewfile.lock.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (tomrf1) [inline scripts/nginx/Brewfile.lock.json:1]: this can also be deleted

File: src/web/components/SlotBodyEnd/ReaderRevenueEpic.tsx [hunk 3]
```diff
@@ -175,7 +182,7 @@ export const canShow = async ({
 
 	const response = await getBodyEnd(
 		contributionsPayload,
-		`${contributionsServiceUrl}/epic${queryString}`,
+		`http://localhost:8082/epic${queryString}`,
```
Comment (TomPretty) [inline src/web/components/SlotBodyEnd/ReaderRevenueEpic.tsx:185]: This is always really fiddly as we need it when running locally, but we'll don't want this change

File: src/web/lib/contributions.ts [hunk 4]
```diff
@@ -151,7 +148,7 @@ export const getArticleCountConsent = (): Promise<boolean> => {
 				if (!hasRequiredConsents) {
 					removeArticleCountsFromLocalStorage();
 				}
-
+				console.log({ hasRequiredConsents });
```
Comment (TomPretty) [inline src/web/lib/contributions.ts:151]: we can delete this too!

File: src/web/components/App.tsx [hunk 5]
```diff
@@ -227,7 +227,7 @@ export const App = ({ CAPI, NAV, ophanRecord }: Props) => {
 	// article pages when other pages are supported by DCR.
 	useEffect(() => {
 		const incrementArticleCountsIfConsented = async () => {
-			if (await getArticleCountConsent()) {
+			if (await hasOptedOutOfArticleCount()) {
```
Comment (tomrf1) [inline src/web/components/App.tsx:230]: just noticed this now needs a `!`
```
const hasOptedOut = await hasOptedOutOfArticleCount();
if (!hasOptedOut) {
```

File: src/web/components/StickyBottomBanner/ReaderRevenueBanner.tsx [hunk 6]
```diff
@@ -149,7 +150,7 @@ export const canShowRRBanner: CanShowFunctionType = async ({
 	}
 
 	const countryCode = await asyncCountryCode;
-	const hasConsentedToArticleCounts = await getArticleCountConsent();
+	const hasConsentedToArticleCounts = await hasCmpConsentForArticleCount();
```
Comment (tomrf1) [inline src/web/components/StickyBottomBanner/ReaderRevenueBanner.tsx:153]: This actually needs to call `hasOptedOutOfArticleCount`.
Also, the existing name `hasConsentedToArticleCounts` is not helpful here, as it later gets assigned to a different name when we build the payload:
`hasOptedOutOfArticleCount: !hasConsentedToArticleCounts`

Let's just rename `hasConsentedToArticleCounts` to `hasOptedOutOfArticleCount` in this file, and we can avoid using the `!` operator so much

File: src/web/lib/contributions.ts [hunk 7]
```diff
@@ -127,19 +127,19 @@ export const shouldHideSupportMessaging = (
...
 };
 
-export const getArticleCountConsent = (): Promise<boolean> => {
-	if (hasOptedOutOfArticleCount()) {
-		return Promise.resolve(false);
-	}
+export const hasCmpConsentForArticleCount = (): Promise<boolean> => {
 	return new Promise((resolve) => {
+		if (getCookie('gu-cmp-disabled')) {
+			resolve(true);
```
Comment (tomrf1) [inline src/web/lib/contributions.ts:141]: this prevents the cypress tests from breaking, as `onConsentChange` doesn't work

File: cypress/integration/parallel-3/article.interactivity.spec.js [hunk 8]
```diff
@@ -54,7 +54,7 @@ describe('Interactivity', function () {
 				cy.contains('Lifestyle');
 				cy.get('[data-component="most-popular"]').scrollIntoView({
 					duration: 300,
-					offset: { top: -30 },
+					offset: { top: 50 },
```
Comment (tomrf1) [inline cypress/integration/parallel-3/article.interactivity.spec.js:57]: this test is flakey - it needs to scroll down enough to trigger the request

**Review by tomrf1 — APPROVED** [review]
great!

**Pull request merged**